TechTalk.JiraRestClient/JiraClient.cs

```diff
@@ -14,29 +14,36 @@ namespace TechTalk.JiraRestClient
 
     public class JiraClient<TIssueFields> : IJiraClient<TIssueFields> where TIssueFields : IssueFields, new()
     {
-        private readonly string username;
-        private readonly string password;
-        private readonly JsonDeserializer deserializer;
-        private readonly string baseApiUrl;
-        public JiraClient(string baseUrl, string username, string password)
+        private readonly string _username;
+        private readonly string _password;
+        private readonly JsonDeserializer _deserializer;
+        private readonly string _baseApiUrl;
+
+        public JiraClient(string baseUrl, string username, string password, int maxItemsPerRequest = 50)
         {
-            this.username = username;
-            this.password = password;
-            
-            baseApiUrl = new Uri(new Uri(baseUrl), "rest/api/2/").ToString();
-            deserializer = new JsonDeserializer();
+            _username = username;
+            _password = password;
+            MaxItemsPerRequest = maxItemsPerRequest;
+
+            _baseApiUrl = new Uri(new Uri(baseUrl), "rest/api/2/").ToString();
+            _deserializer = new JsonDeserializer();
         }
 
+        /// <summary>
+        /// Maximum items that should be returned with each request.
+        /// </summary>
+        public int MaxItemsPerRequest { get; set; }
```

Comment:
Either add property to the interface or remove it (I'd suggest the second option)